# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/network/ConnectionWorker.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/network/ConnectionWorker.cpp
--- a/dom/network/ConnectionWorker.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/network/ConnectionWorker.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -120,7 +120,7 @@
  public:
   NotifyRunnable(WorkerPrivate* aWorkerPrivate, ConnectionProxy* aProxy,
                  ConnectionType aType, bool aIsWifi, uint32_t aDHCPGateway)
-      : WorkerThreadRunnable(aWorkerPrivate, "NotifyRunnable"),
+      : WorkerThreadRunnable("NotifyRunnable"),
         mProxy(aProxy),
         mConnectionType(aType),
         mIsWifi(aIsWifi),
@@ -183,7 +183,7 @@
       new NotifyRunnable(mWorkerRef->Private(), this,
                          static_cast<ConnectionType>(aNetworkInfo.type()),
                          aNetworkInfo.isWifi(), aNetworkInfo.dhcpGateway());
-  runnable->Dispatch();
+  runnable->Dispatch(mWorkerRef->Private());
 }
 
 void ConnectionProxy::Shutdown() {